hashes/src/siphash24.rs

```diff
@@ -211,12 +210,57 @@ impl Hash {
     }
 
     /// Returns the (little endian) 64-bit integer representation of the hash value.
-    pub fn as_u64(&self) -> u64 { u64::from_le_bytes(self.0) }
+    pub fn as_u64(&self) -> u64 {
+        u64::from_le_bytes(self.0)
+    }
 
     /// Creates a hash from its (little endian) 64-bit integer representation.
-    pub fn from_u64(hash: u64) -> Hash { Hash(hash.to_le_bytes()) }
+    pub fn from_u64(hash: u64) -> Hash {
+        Hash(hash.to_le_bytes())
+    }
+
+    fn from_slice(sl: &[u8]) -> Result<Self, crate::FromSliceError> {
+        let mut ret = [0; 8];
+        ret.copy_from_slice(sl);
+        Ok(Hash(ret))
+    }
+}
+
+impl crate::Hash for Hash {
+    type Bytes = [u8; 8];
+
+    const LEN: usize = 8;
+    const DISPLAY_BACKWARD: bool = false;
+
+    fn from_slice(sl: &[u8]) -> Result<Self, crate::FromSliceError> {
+        Self::from_slice(sl)
+    }
+
+    fn to_byte_array(self) -> Self::Bytes {
+        self.0
+    }
+
+    fn as_byte_array(&self) -> &Self::Bytes {
+        &self.0
+    }
+
+    fn from_byte_array(bytes: Self::Bytes) -> Self {
+        Hash(bytes)
+    }
 }
 
+impl<I: SliceIndex<[u8]>> Index<I> for Hash {
+    type Output = I::Output;
+
+    #[inline]
+    fn index(&self, index: I) -> &Self::Output {
+        &self.0[index]
+    }
+}
```

Comment:
Oh, we still have `Index`? We probably shouldn't.

Reply:
Agreed, though I also wouldn't mind leaving this for another PR. Possibly people are using this and we would need to have a discussion.

Reply:
Of course.

Reply:
#3115